cmd/validate-yaml/main.go

```diff
@@ -0,0 +1,114 @@
+package validate_yaml
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	pkgExtractor "github.com/datreeio/datree/pkg/extractor"
+	"github.com/datreeio/datree/pkg/utils"
+	"github.com/spf13/cobra"
+)
+
+type IPrinter interface {
+	PrintFilename(title string)
+	PrintYamlValidationErrors(validationErrors []error)
+	PrintYamlValidationSummary(passedFiles int, allFiles int)
+	PrintMessage(messageText string, messageColor string)
+}
+
+type IReader interface {
+	FilterFiles(paths []string) ([]string, error)
+}
+
+type IExtractor interface {
+	ExtractConfigurationsFromYamlFile(path string) (*[]pkgExtractor.Configuration, string, *pkgExtractor.InvalidFile)
+}
+
+type ValidateYamlCommandContext struct {
+	Printer   IPrinter
+	Reader    IReader
+	Extractor IExtractor
+}
+
+var YamlNotValidError = errors.New("")
+
+func SetSilentMode(cmd *cobra.Command) {
+	cmd.SilenceUsage = true
+	cmd.SilenceErrors = true
+}
+
+func New(ctx *ValidateYamlCommandContext) *cobra.Command {
+	return &cobra.Command{
+		Use:   "validate-yaml <files>",
+		Short: "validates yaml files structure",
+		Long:  "Validates yaml files <files> structure.",
+		Example: utils.Example(`
+		# Validate yaml files using file path
+		datree validate-yaml kube-prod/deployment.yaml
+
+		# Validate yaml files using glob pattern
+		datree validate-yaml kube-*/*.yaml
+
+		# Validate yaml files by sending manifests through stdin
+		cat kube-prod/deployment.yaml | datree validate-yaml -
+		`),
+		Hidden: true,
+		Args: func(cmd *cobra.Command, args []string) error {
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			SetSilentMode(cmd)
+			var err error = nil
+
+			defer func() {
+				if err != nil {
+					ctx.Printer.PrintMessage(strings.Join([]string{"\n", err.Error(), "\n"}, ""), "error")
+				}
+			}()
+
+			filesPaths, err := ctx.Reader.FilterFiles(args)
+			if err != nil {
+				return err
+			}
+
+			filesCount := len(filesPaths)
+			if filesCount == 0 {
+				err = fmt.Errorf("No files detected")
+				return err
+			}
+
+			invalidYamlFiles := ValidateFiles(ctx.Extractor, filesPaths)
+			PrintValidationResults(ctx.Printer, invalidYamlFiles, filesCount)
+
+			if len(invalidYamlFiles) > 0 {
+				return YamlNotValidError
+			}
+
+			return nil
+		},
+	}
+}
+
+func ValidateFiles(extractor IExtractor, filesPaths []string) []*pkgExtractor.InvalidFile {
```

Comment:
looks to me that this logic can be in a `yamlValidator` package (or any other name for that matter).
you can seperate the command implementation from the yaml validation step. 
this will allow us to use the same package for both `datree test` and `datree validate-yaml` should we decide to

Reply:
sounds good